py4DSTEM/process/phase/iterative_multislice_ptychography.py

```diff
@@ -420,6 +414,13 @@ def preprocess(
                 )
             )
 
+        if self._positions_mask is not None and self._positions_mask.dtype != "bool":
+            warnings.warn(
+                ("`positions_mask` converted to `bool` array"),
+                UserWarning,
+            )
+            self._positions_mask = np.asarray(self._positions_mask, dtype="bool")
+
```

Comment:
A lil tangential to this, but this code has been replaced in multiple places, is it possible to update base classes?